examples/Finding_Local_Peaks_in_Solar_Data.py

```diff
@@ -0,0 +1,76 @@
+"""
+=================================
+Finding Local Peaks in Solar Data
+=================================
+
+How to find regions of local maxima
+"""
+import astropy.units as u
+from skimage.feature import peak_local_max
+import matplotlib.pyplot as plt
+from mpl_toolkits import mplot3d
+from mpl_toolkits.mplot3d import Axes3D
+import numpy as np
+
+import sunpy.map
+from sunpy.data.sample import AIA_193_IMAGE
+
+###############################################################################
+# We will first create a Map using some sample data and display it.
+
+aiamap = sunpy.map.Map(AIA_193_IMAGE)
+plt.figure()
+aiamap.plot()
+plt.colorbar()
+
+
+###############################################################################
+# Before we find regions of local maxima, we need to create some variables that
+# store pixel coordinates for the 2D SDO/AIA data we are using.
+# These variables are used for plotting in 3D later on.
+
+x = y = np.arange(0, len(aiamap.data))
+X, Y = np.meshgrid(x, y)
+
+
+###############################################################################
+# We will only consider peaks within the AIA data that are above a threshold
+# value. Once these values are thresholded, the next step is to
+# calcualte the pixel locations of local maxima posistions.
+# This function comes from sci-kit image and the documenation is found
+# here `~skimage.feature.peak_local_max`.
+
+coordinates = peak_local_max(aiamap.data, min_distance=60, threshold_rel=0.2)
+
+
+###############################################################################
+# We now check for the indices at which we get such a local maxima and plot
+# those positions marked red in the aiamap data.
+
+fig = plt.figure(figsize=(12, 8))
+ax = fig.add_subplot(111, projection='3d')
+ax.plot_surface(X, Y, aiamap.data)
+ax.view_init(elev=39, azim=64)
+zs2 = np.array([aiamap.data[x][y] for x, y in zip(coordinates[:, 0], coordinates[:, 1])])
```

Comment:
I might be mis-understanding this but is this not the same as:
```python
zs2 = aiamap.data[coordinates]
```

Also `zs2` is not a very descriptive variable name.

Reply:
`coordinates` is a 2d numpy array.
So first axis = X value , second axis = Y value.

So I guess `aiamap.data[coordinates]` returns combination of all X and Y values instead of going row by row which ensured in the implementation of `zs2`

`aiamap.data[coordinates]` returns

` array([[[  8.07000000e+02,   7.95000000e+02,   8.74000000e+02, ...,
           6.41000000e+02,   6.36000000e+02,   6.09000000e+02],
        [  7.80000000e+01,   7.90000000e+01,   7.90000000e+01, ...,
           1.36000000e+02,   1.16000000e+02,   8.20000000e+01]],

       [[  8.25000000e+02,   7.81000000e+02,   8.55000000e+02, ...,
           6.67000000e+02,   6.74000000e+02,   6.23000000e+02],
        [  2.55000000e+02,   2.68000000e+02,   2.52000000e+02, ...,
           3.36000000e+02,   3.41000000e+02,   3.04000000e+02]],

       [[  1.05300000e+03,   1.07800000e+03,   1.05900000e+03, ...,
           8.24000000e+02,   8.13000000e+02,   7.75000000e+02],
        [  4.00000000e+00,   1.00000000e+00,   8.00000000e+00, ...,
          -1.00000000e+00,   5.00000000e+00,   1.00000000e+01]],

       [[  1.02700000e+03,   1.00500000e+03,   1.01700000e+03, ...,
           8.41000000e+02,   7.91000000e+02,   7.73000000e+02],
        [  1.23800000e+03,   1.20200000e+03,   1.23800000e+03, ...,
           6.32000000e+02,   6.15000000e+02,   6.27000000e+02]],

       [[  1.04100000e+03,   1.07400000e+03,   1.15400000e+03, ...,
           8.37000000e+02,   8.91000000e+02,   8.01000000e+02],
        [  9.40000000e+02,   9.91000000e+02,   9.71000000e+02, ...,
           8.18000000e+02,   8.56000000e+02,   7.54000000e+02]],

       [[  6.68000000e+02,   6.65000000e+02,   6.72000000e+02, ...,
           4.37000000e+02,   4.59000000e+02,   4.71000000e+02],
        [  1.61000000e+02,   1.52000000e+02,   1.74000000e+02, ...,
           2.51000000e+02,   2.26000000e+02,   2.29000000e+02]]], dtype=float32)
In [4]:`

while 
`zs2` returns

`array([  99807.,  125968.,  103099.,   91572.,   71012.,  260644.], dtype=float32)
In [5]:`

Reply:
There must be a way to use it without the for loop.
What about `aiamap.data[coordinates[;,0], coordinates[:,1]]`?

Reply:
works fine !!

I don't know why it didn't occur to me then